tinycrypt/ecc.c

```diff
@@ -1037,82 +1167,88 @@ int uECC_generate_random_int(uECC_word_t *random, const uECC_word_t *top,
 }
 
 
-int uECC_valid_point(const uECC_word_t *point, uECC_Curve curve)
+int uECC_valid_point(const uECC_word_t *point)
 {
 	uECC_word_t tmp1[NUM_ECC_WORDS];
 	uECC_word_t tmp2[NUM_ECC_WORDS];
-	wordcount_t num_words = curve->num_words;
+	wordcount_t num_words = NUM_ECC_WORDS;
+	volatile uECC_word_t diff = -1u;
```

Comment:
VS 2013 doesn't like this line
`warning C4146: unary minus operator applied to unsigned type, result still unsigned`